compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeBuilder.java

```diff
@@ -3713,29 +3713,31 @@ public BLangNode transform(QueryExpressionNode queryExprNode) {
 
     public BLangNode transform(OnFailClauseNode onFailClauseNode) {
         Location pos = getPosition(onFailClauseNode);
-        BLangSimpleVariableDef variableDefinitionNode = (BLangSimpleVariableDef) TreeBuilder.
-                createSimpleVariableDefinitionNode();
-        BLangSimpleVariable var = (BLangSimpleVariable) TreeBuilder.createSimpleVariableNode();
-        boolean isDeclaredWithVar = onFailClauseNode.typeDescriptor().kind() == SyntaxKind.VAR_TYPE_DESC;
-        var.isDeclaredWithVar = isDeclaredWithVar;
-        if (!isDeclaredWithVar) {
-            var.setTypeNode(createTypeNode(onFailClauseNode.typeDescriptor()));
-        }
-        var.pos = getPosition(onFailClauseNode);
-        var.setName(this.createIdentifier(onFailClauseNode.failErrorName()));
-        var.name.pos = getPosition(onFailClauseNode.failErrorName());
-        variableDefinitionNode.setVariable(var);
-        variableDefinitionNode.pos = getPosition(onFailClauseNode.typeDescriptor(), onFailClauseNode.failErrorName());
-
-
         BLangOnFailClause onFailClause = (BLangOnFailClause) TreeBuilder.createOnFailClauseNode();
         onFailClause.pos = pos;
-
-        onFailClause.isDeclaredWithVar = isDeclaredWithVar;
-        markVariableWithFlag(variableDefinitionNode.getVariable(), Flag.FINAL);
-        onFailClause.variableDefinitionNode = variableDefinitionNode;
+        onFailClauseNode.typeDescriptor().ifPresent(typeDescriptorNode -> {
+            BLangSimpleVariableDef variableDefinitionNode = (BLangSimpleVariableDef) TreeBuilder.
+                    createSimpleVariableDefinitionNode();
```

Comment:
Would it be better to keep `(BLangSimpleVariableDef) TreeBuilder.createSimpleVariableDefinitionNode()` as a single line?